node/core/pvf/src/prepare/worker.rs

```diff
@@ -288,31 +284,43 @@ pub fn worker_entrypoint(socket_path: &str) {
 				worker_pid = %std::process::id(),
 				"worker: preparing artifact",
 			);
-			let artifact_bytes = prepare_artifact(&code).serialize();
 
-			// Write the serialized artifact into into a temp file.
-			tracing::debug!(
-				target: LOG_TARGET,
-				worker_pid = %std::process::id(),
-				"worker: writing artifact to {}",
-				dest.display(),
-			);
-			async_std::fs::write(&dest, &artifact_bytes).await?;
+			let result = match prepare_artifact(&code) {
+				Artifact::Error(err) => {
+					// Serialized error will be written into the socket.
+					Err(err)
+				},
+				Artifact::Compiled(compiled_artifact) => {
+					// Write the serialized artifact into a temp file.
+					// Since a compiled artifact can be heavy, we send an empty
+					// `Ok` to indicate the success.
```

Comment:
This is a bit misleading. In reality, we do not put the artifact on UDS since the host is not expected to hold them in memory. That is by design since the artifacts can be quite hefty. The difference is subtle, but it is important to not oversimplify.

Here, if the reader already knows about this high-level design point, then they will be just distracted and no benefit will be gained. OTOH, if the reader is not familiar with it, then I do not think this comment will help understand it.

Generally, as we write comments, we should strive to be as useful to the reader as possible. We should not forget that comments have costs too: they require maintenance when things change and they require attention when read. 

I see two alternatives here:

1. Remove the whole thing
2. Describe what we are doing here instead of why.